deployer/src/deployer/upload.py

```diff
@@ -508,6 +518,13 @@ def upload_content(build_directory, content_roots, config):
     show_progress_bar = not config["no_progressbar"]
     upload_redirects = not config["no_redirects"]
     prune = config["prune"]
+    archived_txt_file = config["archived_files"]
+
+    archived_files = set()
+    if archived_txt_file:
+        archived_files.update(parse_archived_txt_file(archived_txt_file))
+        if not archived_files:
+            raise Exception(f"found no entries inside {archived_txt_file}")
```

Comment:
Assuming we make `parse_archived_txt_file` a generator, this becomes a super-quick, short-circuit-style check, and we get rid of having to create a `list` (the result of the `parse_archived_txt_file` function) and a `set` (`archived_files`) that are just used as intermediate steps towards creating `archived_files_as_keys` further below:
```suggestion
    if archived_txt_file and not any(parse_archived_txt_file(archived_txt_file)):
        raise Exception(f"found no entries inside {archived_txt_file}")
```

and we make one more change (see below).